url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/utilities/DerbyResults.java

```diff
@@ -0,0 +1,565 @@
+/*   
+   Copyright 2011-2013 The Cassandra Consortium (cassandra-fp7.eu)
+
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package eu.cassandra.sim.utilities;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+
+// TODO: Auto-generated Javadoc
+/**
+ * The wrapper class for storing simulation output results in an Apache Derby database.
+ *
+ * @author Fani A. Tzima (fani [dot] tzima [at] iti [dot] gr)
+ */
+public class DerbyResults implements DBResults{
+	
+	/** The database name. */
+	private String dbname;
+	
+	/** The database connection. */
+	private Connection conn;
+	
+	/**
+	 * Instantiates a new DerbyResults wrapper object.
+	 *
+	 * @param adbname the target database name
+	 */
+	public DerbyResults(String adbname) {
+ 		dbname = adbname;
+	}
+	
+	/* (non-Javadoc)
+	 * @see eu.cassandra.sim.utilities.DBResults#createTablesAndIndexes()
+	 */
+	@Override
+	public void createTablesAndIndexes() {
+		conn = getConnection();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *conn*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).